### PR TITLE
Several warning squashes

### PR DESCRIPTION
**Pull request by pavanbalaji**
## Pull Request Description

The previous warning tests in yaksa were basically bogus.  The strict option was missing the optimization flag (which disabled some warnings), and the setting of the compiler to `gcc-4` and `gcc-9` was completely bogus (it was never getting set in the jenkins script).  Plus, there was no testing with other compilers.  This PR was done in conjunction with improving the jenkins test for better warning checking.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

**Comment by pavanbalaji** [comment]
test:yaksa/warnings

File: maint/clmake [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+##
+## Copyright (C) by Argonne National Laboratory
+##     See COPYRIGHT in top-level directory
+##
+
+# Set Defaults
+$debug = 0;
+$debugDir = 0;
+$gDebugOther = 0;
+
```
Comment (gcongiu) [inline maint/clmake:11]: Can you add a comment in the commit describing what this script does and why is needed?

Comment (pavanbalaji) [reply maint/clmake:11]: Done.

**Pull request merged**